### ISSUE TITLE
Brokerable (non-ATO'd) Solr instances available to data.gov team via cloud.gov

### ISSUE THREAD

**Issue by mogul** [GSA/data.gov#1808]
## User Story

In order to get access to Solr instances on-demand for use in our cloud.gov environments, the data.gov team wants to operate a broker that makes brokered Solr instances available.

## Acceptance Criteria
- [x] GIVEN I run `cf login -a api.fr.cloud.gov --sso` and authenticate \
  AND I run `cf target -o gsa-datagov -s <development|staging|production>`  \
  WHEN I run `cf marketplace` \
  THEN I see the datagov-solr broker's services and plans listed
- [x] GIVEN I run `cf login -a api.fr.cloud.gov --sso` and authenticate \
  AND I run `cf target -o gsa-datagov -s <development|staging|production>`  \
  WHEN I run `cf create-service solr-cloud base datagov-solr` \
  THEN I see the instance is running
  AND I can `cf bind-service <app-name> <instance-name>` \
  AND I can see credentials for the Solr instance in `cf env <app-name>`

## Background
This is a follow-up to [the story about being able to use Terraform to spin up Helm charts in AWS EKS](https://github.com/GSA/datagov-deploy/issues/1805). The goal here is to have an operational broker being deployed via CI. 

## Security Considerations ([required](https://nvd.nist.gov/800-53/Rev4/control/CM-4))
[comment]: # "Our SSP says 'The Data.gov team ensures security implications are considered as part of the agile requirements refinement process by including a section in the issue template used as a basis for new work.' so please don't remove this section without care."
The broker itself is intended to be a simple public-facing web service deployed on cloud.gov, requiring credentials to be able to register and access it. All the usual CI, scanning, and credential-handling processes should be in place so that this new service conforms to our Configuration Management Plan (CM plan) and is therefore ready for submission to A&A for ATO.

## Sketch
- Get [this PR merged](https://github.com/GSA/catalog-solr/pull/1), if possible
- Bundle up the necessary terraform using [brokerpak](https://github.com/GoogleCloudPlatform/gcp-service-broker/blob/master/docs/brokerpak-intro.md)
- Using CircleCI (and ideally Terraform), [deploy the GCP broker](https://github.com/cloud-gov/cf-byo-broker/tree/master/gcp-service-broker) with that Solr brokerpak configured
- Celebrate, big-time!

**Comment by mogul** [comment]
The Solr brokerpak should contain the following terraform:
  - Use Terraform's AWS EKS provider to provision an EKS cluster
  - Use Terraform's Helm provider to deploy the [Solr Operator chart](https://github.com/bloomberg/solr-operator/tree/master/helm/solr-operator).

Open questions: 

- How does the actual SolrCloud instance get [provisioned through the Solr Operator](https://github.com/bloomberg/solr-operator/blob/master/docs/local_tutorial.md#start-an-example-solr-cloud-cluster)?
  - Can this be done in Terraform?
    - Yes, see the [ExternalDNS `kubernetes_deployment` example](https://www.hashicorp.com/blog/using-the-kubernetes-and-helm-providers-with-terraform-0-12/)

**Comment by mogul** [comment]
Stuck on the CF Terraform provider timing out at 15 minutes before the brokered EKS back-end for the SolrCloud broker is fully provisioned. I've [opened a PR upstream](https://github.com/cloudfoundry-community/terraform-provider-cloudfoundry/pull/314) bu there is some [confusion and pushback](https://github.com/cloudfoundry-community/terraform-provider-cloudfoundry/pull/314#issuecomment-797002342).

**Comment by mogul** [comment]
Rollout blocked on [a bug in the CF provider](https://github.com/cloudfoundry-community/terraform-provider-cloudfoundry/issues/329). The author is responsive and [I'm helping test a fix PR](https://github.com/cloudfoundry-community/terraform-provider-cloudfoundry/pull/333#issuecomment-811304253), though without luck yet. 😞

**Comment by mogul** [comment]
Upstream bug fixed! Just testing the AC before I call this Done.

**Comment by mogul** [comment]
The end-to-end flow for provisioning and binding Solr instances and getting credentials has been tested as working. However, the Solr instances that are provisioned are stuck in a Pending state in AWS Fargate. I'm opening a new issue to resolve that.
